source/geometry_model/spherical_shell.cc

```diff
@@ -45,22 +46,170 @@ namespace aspect
                                "this geometry are 90, 180, and 360 in 2d; "
                                "and 90 and 360 in 3d."));
 
+      // Custom mesh extrusion is only implemented for full spherical shells
+      // for now, so check that custom mesh schemes are only being used when
+      // the opening angle is 360 degrees.
+      AssertThrow (phi == 360 || (custom_mesh == none),
+                   ExcMessage ("The only opening angle that is allowed for "
+                               "this geometry with a custom mesh is 360."));
+
       if (phi == 360)
         {
-          GridGenerator::hyper_shell (coarse_grid,
-                                      Point<dim>(),
-                                      R0,
-                                      R1,
-                                      (n_cells_along_circumference == 0
-                                       ?
-                                       // automatic choice that leads to reasonable
-                                       // meshes with the typical aspect ratio of
-                                       // the Earth
-                                       (dim==3 ? 96 : 12)
-                                       :
-                                       // user choice
-                                       n_cells_along_circumference),
-                                      true);
+          if (custom_mesh == none)
+            {
+              // If we are not using a custom mesh scheme, the mesh is generated
+              // as per the original code.
+              GridGenerator::hyper_shell (coarse_grid,
+                                          Point<dim>(),
+                                          R0,
+                                          R1,
+                                          (n_cells_along_circumference == 0
+                                           ?
+                                           // automatic choice that leads to reasonable
+                                           // meshes with the typical aspect ratio of
+                                           // the Earth
+                                           (dim==3 ? 96 : 12)
+                                           :
+                                           // user choice
+                                           n_cells_along_circumference),
+                                          true);
+            }
+          else
+            {
+              // If we are using a custom mesh scheme, we need to create
```

Comment:
can you please add ``Assert(custom_mesh == slices || custom_mesh == list, ExcNotImplemented());``. That makes sure if somebody adds a new option, this will trigger.